bouffalo-hal/src/sdio.rs

```diff
@@ -66,151 +65,3473 @@ pub struct RegisterBlock {
+        Self((self.0 & !Self::CMD_TYPE) | (Self::CMD_TYPE & ((val as u16) << 6)))
+    }
+    /// Get command type.
+    #[inline]
+    pub const fn cmd_type(self) -> CmdType {
+        match (self.0 & Self::CMD_TYPE) >> 6 {
+            1 => CmdType::Suspend,
+            2 => CmdType::Resume,
+            3 => CmdType::Abort,
+            _ => CmdType::Other,
+        }
+    }
+    /// Set this bit to 1 to indicate that data is present and shall be transferred using the DAT line.
+    #[inline]
+    pub const fn set_data_present(self) -> Self {
+        Self((self.0 & !Self::DATA_PRSENT) | (Self::DATA_PRSENT & (1 << 5)))
+    }
+    /// Set this bit to 0 for the following:
+    /// (1) Commands using only CMD line (ex.CMD52).
+    /// (2) Commands with no data transfer but using busy signal on DAT[0] line (R1b or R5b ex. CMD38).
+    /// (3) Resume command.
+    #[inline]
+    pub const fn unset_data_present(self) -> Self {
+        Self((self.0 & !Self::DATA_PRSENT) | (Self::DATA_PRSENT & (0 << 5)))
+    }
+    /// If Data Present bit is set.
+    #[inline]
+    pub const fn is_data_present(self) -> bool {
+        (self.0 & Self::DATA_PRSENT) >> 5 == 1
+    }
+    /// Enable check the Index field.
+    #[inline]
+    pub const fn enable_index_check(self) -> Self {
+        Self((self.0 & !Self::CMD_INDEX_CHECK) | (Self::CMD_INDEX_CHECK & (1 << 4)))
+    }
+    /// Disable check the Index field.
+    #[inline]
+    pub const fn disable_index_check(self) -> Self {
+        Self((self.0 & !Self::CMD_INDEX_CHECK) | (Self::CMD_INDEX_CHECK & (0 << 4)))
+    }
+    /// If check the Index field is enabled.
+    #[inline]
+    pub const fn is_index_check_enabled(self) -> bool {
+        (self.0 & Self::CMD_INDEX_CHECK) >> 4 == 1
+    }
+    /// Enable command crc.
+    #[inline]
+    pub const fn enable_cmd_crc(self) -> Self {
+        Self((self.0 & !Self::CMD_CRC) | (Self::CMD_CRC & (1 << 3)))
+    }
+    /// Disable command crc.
+    #[inline]
+    pub const fn disable_cmd_crc(self) -> Self {
+        Self((self.0 & !Self::CMD_CRC) | (Self::CMD_CRC & (0 << 3)))
+    }
+    /// If command crc is enabled.
+    #[inline]
+    pub const fn is_cmd_crc_enabled(self) -> bool {
+        (self.0 & Self::CMD_CRC) >> 3 == 1
+    }
+    /// Set Response Type.
+    #[inline]
+    pub const fn set_response_type(self, val: ResponseType) -> Self {
+        Self((self.0 & !Self::RESPONSE_TYPE) | (Self::RESPONSE_TYPE & (val as u16)))
+    }
+    /// Get Response Type.
+    #[inline]
+    pub const fn response_type(self) -> ResponseType {
+        match self.0 & Self::RESPONSE_TYPE {
+            1 => ResponseType::ResponseLen136,
+            2 => ResponseType::ResponseLen48,
+            3 => ResponseType::ResponseLen48Check,
+            _ => ResponseType::NoResponse,
+        }
+    }
+}
+
+/// Register that stores responses from SD cards.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 #[repr(transparent)]
 pub struct Response(u128);
```

Comment:
We should check if reading and writing of `u128` values on Rust are supported in the target peripheral later.